### PR TITLE
[12.0][MIG] bi_view_editor

### PR DESCRIPTION
**Pull request by astirpe**
This PR is the porting of `bi_view_editor` to V12 with the following improvements:

- [x] Store the JSON data structure in ORM
- [x] Improved robustness
- [x] Fix of fields/models translation
- [x] Add check for consistency of groups
- [x] Development of new tests
- [x] Add LEFT JOIN capabilities
- [x] Add sums and avg capabilities for tree views
- [x] Provide graph view for table relations

### CODE REVIEW

File: bi_view_editor/models/ir_model.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+            related_fields = self.get_related_fields(model_ids)
+            list_id = _get_list_id(model_ids, related_fields)
+            list_model = _get_list_relation(related_fields)
+            domain += ['|',
+                       ('id', 'in', list_id),
+                       ('model', 'in', list_model)]
+        return self.sudo().search(domain)
+
+    @api.model
+    def get_models(self, model_ids=None):
```
Comment (MiquelRForgeFlow) [inline bi_view_editor/models/ir_model.py]: `model_ids={}` instead

**Review by MiquelRForgeFlow — APPROVED** [review]
code review :+1:

File: bi_view_editor/models/ir_model.py (hunk 1 above)
Comment (astirpe) [reply bi_view_editor/models/ir_model.py]: This proposed change causes a warning:

```
************* Module bi_view_editor.models.ir_model
bi_view_editor/models/ir_model.py:173: [W0102(dangerous-default-value), IrModel.get_models] Dangerous default value {} as argument

```

so I will revert it back to `model_ids=None`.

**Comment by astirpe** [comment]
I'm adding tests but the **codecov/patch** is not increasing. The coverage is actually much higher, around 90%: https://codecov.io/gh/OCA/reporting-engine/tree/e99acf4ad56455afe6574307ac9712a3ed429cf6/bi_view_editor

**Comment by astirpe** [comment]
@pedrobaeza do you know why the codecov/patch percentage does not increase?

**Comment by patel1008b** [comment]
@astirpe @pedrobaeza @mreficent what do you think it will merge in the main branch. It's Badly required for one of project.

Thanks for this PR.

**Comment by astirpe** [comment]
@bhaveshselarka this PR is not yet merged because of one extra review is still missing. You may want to test this PR on Runbot and report here the results. That would speed up the approval process. Thanks!

**Comment by mreficent** [comment]
That monkey patch of the last commit is not obsolete, I think.

**Comment by MiquelRForgeFlow** [comment]
I have a tested two migrations from v11. Without last commit, no problem; with last commit, it breaks. Please, remove last commit. That monkey patch of the last commit is not obsolete.

**Comment by astirpe** [comment]
Thank you @mreficent !

**Comment by patel1008b** [comment]
@astirpe @pedrobaeza @mreficent what do you think it will merge in the main branch. It's Badly required for one of project.

Thanks for this PR.

**Comment by pedrobaeza** [comment]
@bhaveshselarka why don't you review the PR, try on runbot and give your feedback for speeding up the process?

**Comment by patel1008b** [comment]
Yes, @pedrobaeza I think its best idea, Let me do that stuff.

I am at currently: https://runbot.odoo-community.org/runbot/repo/github-com-oca-reporting-engine-143

Where I can find a find this PR's correct instance. Can you please help me out in that?

**Comment by pedrobaeza** [comment]
On the PR itself:
Selección_008
I have launched a rebuild, as the instance has expired. Wait for the completion and you'll be able to access through the blue button. More details in https://odoo-community.org/page/review

**Review by HviorForgeFlow — APPROVED** [review]
LGTM

**Comment by patel1008b** [comment]
@pedrobaeza @hveficent @astirpe @mreficent  will it merge in the main branch ?

**Comment by astirpe** [comment]
@pedrobaeza the codecov is red but the real coverage is much higher, see also my https://github.com/OCA/reporting-engine/pull/293#issuecomment-501274907. There's something wrong with codecov, so for me the actual coverage is more than acceptable.

This PR has two approvals since a long time and I don't think we will get soon any other approvals. Is it okay to merge?

**Comment by pedrobaeza** [comment]
Please squash a bit the commits and I'll proceed to merge. The codecov patch is base on previous commits done, so if you have done several pushes, it can get confused by that.

**Comment by astirpe** [comment]
@pedrobaeza done!

**Comment by astirpe** [comment]
@pedrobaeza would be great if we can get this PR merged. Thank you!

**Pull request merged**